### PR TITLE
Bc 341 sidebar menu

### PR DESCRIPTION
**Pull request by mendo94**
Please see bc-341 Jira ticket for a visual on the sidebar component.

### CODE REVIEW

**Review by HectorIlarraza — CHANGES_REQUESTED** [review]
Overall a great PR Rina this is my first official review of one so I hope I clarified enough on possible suggestions, if not please feel free to hit me up so we can do a walkthrough of it.

File: src/layout/Nav/Nav.tsx [hunk 1]
```diff
@@ -1,46 +1,67 @@
...
+
+interface SidebarProps {
+  isOpen: boolean;
+  toggleSidebar: () => void;
+}
 
-export const Nav: React.FC = () => {
-  const [authLinks, setAuthLinks] = useState<boolean>(false);
+export const Nav = (props: SidebarProps) => {
   const authUser = useSelector(selectAuthUser);
```
[the remark this answers is not in the document]
Comment (mendo94) [reply src/layout/Nav/Nav.tsx:15]: changed authUser._id and used UseApp selector instead.

File: src/index.tsx [hunk 2]
```diff
@@ -2,15 +2,18 @@ import React from 'react';
...
 import reportWebVitals from './reportWebVitals';
 import { Provider } from 'react-redux';
 import { store } from './utilities/redux/store';
 
 const root = ReactDOM.createRoot(document.getElementById('root') as HTMLElement);
 root.render(
-  <Provider store={store}>
-    <App />
-  </Provider>,
+  <Router>
```
[the remark this answers is not in the document]
Comment (mendo94) [reply src/index.tsx:12]: Yes I should have added the router to the index file to begin with, but here we are changing it now

File: src/layout/Nav/Nav.tsx [hunk 3]
```diff
@@ -1,46 +1,67 @@
...
+  };
 
   return (
-    <div className="nav">
-      <nav>
-        <div>
-          <Link to="/sign-up">Sign Up</Link>
+    <nav>
+      <div className="nav-container">
+        {authUser._id !== '' ? (
```
[the remark this answers is not in the document]
Comment (mendo94) [reply src/layout/Nav/Nav.tsx:26]: changed authUser.id to userId

File: src/layout/Sidebar/Sidebar.tsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        <i></i>
+        <i></i>
+        <i></i>
+      </div>
+
+      <div className="current-user">
+        <div className="image"></div>
+        <div>
+          <p className="user-name">
+            {authUser.firstName ? authUser.firstName : 'Wiggly'} {authUser.lastName ? authUser.lastName : 'Jones'}
```
[the remark this answers is not in the document]
Comment (mendo94) [reply src/layout/Sidebar/Sidebar.tsx:41]: then don't be that guy.... changed authUser.first and last to just firstName and lastName.

File: src/layout/Nav/Nav.tsx [hunk 5]
```diff
@@ -1,46 +1,67 @@
-import React, { useEffect, useState } from 'react';
-import { Link } from 'react-router-dom';
 import { useSelector } from 'react-redux';
+import { Link } from 'react-router-dom';
 import { selectAuthUser } from '../../utilities/redux/slices/users/userSlice';
```
[the remark this answers is not in the document]
Comment (mendo94) [reply src/layout/Nav/Nav.tsx:4]: imported the hooks

File: src/layout/Sidebar/Sidebar.tsx [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+interface SidebarProps {
+  isOpen: boolean;
+  toggleSidebar: () => void;
+}
+
+export const Sidebar = (props: SidebarProps) => {
+  const authUser = useSelector(selectAuthUser);
+
+  return (
+    <div className={props.isOpen ? 'sidebar-container active' : 'hide-sidebar'}>
```
[the remark this answers is not in the document]
Comment (mendo94) [reply src/layout/Sidebar/Sidebar.tsx]: Changed isOpen to visibleSidebar for more clarity

File: src/layout/Sidebar/Sidebar.tsx [hunk 7]
```diff
@@ -0,0 +1,51 @@
+import { Link } from 'react-router-dom';
+import { useSelector } from 'react-redux';
+import { selectAuthUser } from '../../utilities/redux/slices/users/userSlice';
```
[the remark this answers is not in the document]
Comment (mendo94) [reply src/layout/Sidebar/Sidebar.tsx]: replaced to the hooks

File: src/layout/Nav/Nav.tsx [hunk 8]
```diff
@@ -1,46 +1,67 @@
...
 import './Nav.scss';
+import Logo from '../../assets/Logo.svg';
+
+interface SidebarProps {
```
[the remark this answers is not in the document]
Comment (mendo94) [reply src/layout/Nav/Nav.tsx]: imported the sidebar from the userslice and used dispatch to handle the function.

**Comment by mendo94** [comment]
Resolved all outstanding errors and ready for second review

**Review by HectorIlarraza — APPROVED** [review]
Excellent Rina! All changes are resolved and approved for the next steps!

**Pull request merged**